### DISCUSSION TITLE
GCNConv GateGraphConv on heterogeneous graphs

### DISCUSSION THREAD

**Discussion by diegocervera** [pyg-team/pytorch_geometric#3489] [Q&A] [answered]
Hi,

Is there a reason why GCNConv, GateGraphConv don't work in bipartite graphs? Is it just a matter of them not being implemented for bipartite graphs or is there an major underlying reason?

Thanks!

**Comment by wsad1** [comment, marked as answer]
[GCNConv](https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html#torch_geometric.nn.conv.GCNConv) expects the source and destination nodes to be of the same type and have the same feature dimension, it cannot support bipartite graphs where source and destination nodes are not of the same type and might have different feature dimension. 

Bottom line, its not a matter of not being implemented, the GCNConv operator was not built to work on bipartite graphs.

**Comment by muety** [comment]
I was asking myself the exact same question, so thanks @diegocervera @wsad1 for clarifying this. However, it's still not entirely clear to me, why GCNConv can't support different feature dimensions. Couldn't Θ simply map from the source node's feature dimension to the one of the destination node?

**Reply by rusty1s** [reply]
We could certainly use disjoint weight matrices for source and destination node features, but this wouldn't align anymore with the original definition of GCN.

**Reply by muety** [reply]
Thanks for the reply! There was a misunderstanding on my end. I was only thinking about message passing from node type B (source) to node type A (destination), where a Θ of size `(d_B, d_A)` would have worked. But of course, messages can be passed in reverse as well along the same type of edges, in which case we'd have to have a second set of weights of size `(d_A, d_B)` Sorry for bothering!
